### PR TITLE
Wrap YAML E2E tests in a function

### PR DESCRIPTION
**Pull request by adrcunha**
This makes the code cleaner, and allow failing faster on error.

### CODE REVIEW

File: test/e2e-tests.sh [hunk 1]
```diff
@@ -65,47 +109,12 @@ kubectl delete buildtemplates --all
...
 
-local failed=0
+failed=0
+
 header "Running Go e2e tests"
 report_go_test -tags e2e ./test/e2e/... -count=1 || failed=1
 
-header "Running YAML e2e tests"
-ko apply -R -f test/ || failed=1
+run_yaml_tests || failed=1
```
Comment (ImJasonH) [inline test/e2e-tests.sh]: Should this be `&& failed=1` ?

If `run_yaml_tests` returns `0`, that'll be interpreted as `false` and therefore trip the `||` branch.

Comment (adrcunha) [reply test/e2e-tests.sh]: No. In UNIX world, 0 means success.

Comment (ImJasonH) [reply test/e2e-tests.sh]: Right. I knew as I was typing it there was no way I had found a real bug. 😄

**Comment by ImJasonH** [comment]
It looks like my recent e2e test change broke the YAML tests, and it wasn't this change. Either way, the YAML tests are effectively skipped. Do you happen to see where my bug is?

**Comment by adrcunha** [comment]
The YAML tests don't seem to be skipped. If you think they are because gubernator only shows "2 tests passed" (the go tests you added), it's because it doesn't parse the output of the shell tests. Unless we intend to keep these tests forever the way they are (instead of rewriting them as go tests), I don't think it's worth updating the script to parse their results.

**Comment by adrcunha** [comment]
Only now I saw your initial comment explaining why you think the tests are being skipped. I don't think they are, because there's a 20s interval between the builds being created and the banner "all tests passed":

```
I0905 22:22:31.546] build "test-workspace-volume" created
I0905 22:22:52.847] **************************************
I0905 22:22:52.847] ***        ALL TESTS PASSED        ***
I0905 22:22:52.847] **************************************
```

However, this is indeed poor logging. Let me fix that in this PR.

**Pull request merged**